python/cudf/cudf/dataframe/column.py

```diff
@@ -159,6 +160,20 @@ def __init__(self, data, mask=None, null_count=None, name=None):
 
         self._update_null_count(null_count)
 
+    def __eq__(self, other):
```

Comment:
I'd expect this to be an elementwise equals instead of a columnwise equals. Was there a reason for adding it under this API?

Reply:
`TypedColumnBase.is_type_equivalent()` calls `_replace_defaults()` to get a dictionary of the default constructor kwargs, then compares them them via `mine == theirs`. Since CategoricalColumn returns the categories in its default kwargs, we have to have some logic to handle comparing the `categories` now that its a Column instead of a tuple.

If `__eq__()` is an elementwise equals, then we'd need to implement `__bool__()` to check its dtype and return `.all()` if it's a bool column, since it seems like the dictionary comparator logic in python is something like `bool(a[key] == b[key])`.

Or we could instead modify the `is_type_equivalent()` method to handle the `categories` key differently than the others.

Reply:
I think we should modify the `is_type_equivalent()` method to use an `equals()` method similar to this: https://github.com/rapidsai/cudf/blob/branch-0.9/python/cudf/cudf/dataframe/series.py#L868. It looks like in cuDF that method doesn't care about the index, but would be good to confirm Pandas is the same. If it doesn't care about the index then we can just move it down to the Column level and utilize it here as well.

Reply:
:+1: updated!